### PR TITLE
feat(basic functionality/settings): add settings pages

### PR DESCRIPTION
**Pull request by STonkoshkur**
feat(settings): add app settings stack
feat(settings/profile management): add update profile and change password pages
feat(navigation/stacks): add full-screen modals stack
chore(store/selectors): update auth selector, add full name and initials props
fix(theming/dependencies): remove react-native-appearance
feat(tests/settings): add e2e tests for  manage profile
chore(tests/utils): add common utils for sign-in/sign-up
docs(readme): update README

### CODE REVIEW

File: src/hooks/useUnsavedFormChangesAlert.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+type UnsavedFormChangesCancellationParams = {
+  title?: string;
+  description?: string;
+  stayOnPageText?: string;
+  discardChangesText?: string;
+  onDiscardChanges?: () => void;
+};
+
+export const useUnsavedFormChangesAlert = (
+  doesFormHasUnsavedChanges: boolean,
```
Comment (Shchepotin) [inline src/hooks/useUnsavedFormChangesAlert.ts:17]: Maybe change isFormHasUnsavedChanges?

File: src/screens/Settings/Index/index.tsx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+  const { colors } = useThemeSchema();
+  const {
+    showPrivacyPolicyModal,
+    showTermsAndConditionsModal,
+  } = useAgreementsModals();
+
+  const dispatch = useDispatch();
+  const authProfile = useSelector(getAuthUserSelector);
+
+  const authProviderName = useMemo(() => authProfile?.provider, [
```
Comment (Shchepotin) [inline src/screens/Settings/Index/index.tsx:48]: I think in this case `useMemo` redundant

Comment (Shchepotin) [reply src/screens/Settings/Index/index.tsx:48]:
```ts
const authProviderName = authProfile?.provider;
```

File: e2e/common/userRegistration.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+const email = `e2e.user.${Date.now()}@example.com`;
```
Comment (Shchepotin) [inline e2e/common/userRegistration.js:1]: Move to function

File: src/hooks/useUnsavedFormChangesAlert.ts (hunk 1 above)
Comment (STonkoshkur) [reply src/hooks/useUnsavedFormChangesAlert.ts:17]: Changed to hasFormUnsavedChanges

File: src/screens/Settings/Index/index.tsx (hunk 2 above)
Comment (STonkoshkur) [reply src/screens/Settings/Index/index.tsx:48]: U r right :)

File: src/components/Card/index.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  style,
+  ...props
+}) => {
+  const { colors } = useThemeSchema();
+
+  return (
+    <View
+      {...props}
+      style={[styles.cardWrapper, { backgroundColor: colors.card }, style]}>
+      <>{children}</>
```
[the remark this answers is not in the document]
Comment (STonkoshkur) [reply src/components/Card/index.tsx:21]: It's needed for future. We need to change View to the Touchable Highlight.

File: e2e/common/userRegistration.js (hunk 3 above)
Comment (STonkoshkur) [reply e2e/common/userRegistration.js:1]: Moved

File: src/components/Toast/index.tsx [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+            backgroundColor: backgroundColor ?? colors.card,
+            borderColor: borderColor ?? colors.border,
+          },
+        ]}>
+        <View
+          style={[
+            styles.iconWrapper,
+            {
+              backgroundColor: iconBackgroundColor,
+              width: iconSize * 1.5,
```
[the remark this answers is not in the document]
Comment (STonkoshkur) [reply src/components/Toast/index.tsx]: moved

File: e2e/common/userRegistration.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+const getRegistrationUserData = () => ({
```
Comment (Shchepotin) [inline e2e/common/userRegistration.js:1]: Maybe rename `getRegistrationUserData` with `generateNewRegistrationUserData`?

File: e2e/common/userRegistration.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  email: `e2e.user.${Date.now()}@example.com`,
+  password: 'secret',
+  firstName: 'User',
+  lastName: `For testing ${Date.now()}`,
+});
+
+const registerUser = async (userData = {}) => {
+  const { email, password, firstName, lastName } = {
+    ...getRegistrationUserData(),
+    ...(userData || {}),
```
Comment (Shchepotin) [inline e2e/common/userRegistration.js:11]: In arguments, you set the default value for `userData`. Can we'll remove `|| {}`?

Comment (STonkoshkur) [reply e2e/common/userRegistration.js:11]: Yep

File: e2e/common/userRegistration.js (hunk 6 above)
Comment (STonkoshkur) [reply e2e/common/userRegistration.js:1]: Yep

**Review by Shchepotin — APPROVED** [review]
LGTM!

**Pull request merged**